### PR TITLE
Trainee recomendation form

### PR DESCRIPTION
**Pull request by ojemuyiwa**

### CODE REVIEW

File: src/app/trainee/revalidation-history.interface.ts [hunk 1]
```diff
@@ -16,6 +16,23 @@ export interface IRecommendation {
   submittedBy: string;
   submissionStatus: string;
   comments: IComment[];
+  deferralReason?: string;
+  deferralDate?: Date;
+
+  // private String proposedOutcomeCode;
```
Comment (omarCreativeDev) [inline src/app/trainee/revalidation-history.interface.ts:22]: is the commented out code WIP due to BE not being complete?

File: src/app/trainee/revalidation-table/revalidation-table.component.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            <th mat-header-cell scope="col" *matHeaderCellDef>Recommendation</th>
+            <td mat-cell *matCellDef="let element">
+                {{ element.recommendation }}
+            </td>
+        </ng-container>
+        <ng-container matColumnDef="outcome">
+            <th mat-header-cell scope="col" *matHeaderCellDef>Outcome</th>
+            <td mat-cell *matCellDef="let element">
+                <div class="outcome" [ngSwitch]="element.outcome">
+                    <mat-icon *ngSwitchCase="'Approved'" aria-hidden="true" class="icon-green">done_all
```
Comment (omarCreativeDev) [inline src/app/trainee/revalidation-table/revalidation-table.component.html:16]: would be good to use an enum or something for the Approved/Rejected/Under review status

File: src/app/trainee/revalidation-table/revalidation-table.component.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                        </div>
+                    </ng-container>
+                </div>
+            </td>
+        </ng-container>
+
+        <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
+        <tr tabindex="0" mat-row *matRowDef="let element; columns: columnsToDisplay" class="element-row"
+            [class.expanded-row]="expandedElement === element" (click)="
+        expandedElement = expandedElement === element ? null : element
```
Comment (omarCreativeDev) [inline src/app/trainee/revalidation-table/revalidation-table.component.html:69]: could this be simplified a little or perhaps thrown in a method as its used several times in this template?

File: src/app/trainee/revalidation-history.interface.ts (hunk 1 above)
Comment (ojemuyiwa) [reply src/app/trainee/revalidation-history.interface.ts:22]: Yeah, I just pasted the back-end properties here to bind after discussions with Uzair and Jay. These are the current JAVA bindings but may be revised. Will delete as appropriate in data-bind task

File: src/app/trainee/revalidation-table/revalidation-table.component.html (hunk 2 above)
Comment (ojemuyiwa) [reply src/app/trainee/revalidation-table/revalidation-table.component.html:16]: Absolutely. No enum or values decided yet for non-gmc related status. Will create enum in anticipation for real values as all these are mocked.

File: src/app/trainee/revalidation-table/revalidation-table.component.html (hunk 3 above)
Comment (ojemuyiwa) [reply src/app/trainee/revalidation-table/revalidation-table.component.html]: done

Comment (ojemuyiwa) [reply src/app/trainee/revalidation-table/revalidation-table.component.html]: done

File: src/app/trainee/revalidation-history.interface.ts (hunk 1 above)
Comment (ojemuyiwa) [reply src/app/trainee/revalidation-history.interface.ts]: removed - kept in notepad

File: src/app/trainee/revalidation-table/revalidation-table.component.html (hunk 2 above)
Comment (ojemuyiwa) [reply src/app/trainee/revalidation-table/revalidation-table.component.html]: done

**Pull request merged**